### PR TITLE
Enabling allow list approach with all image redaction

### PR DESCRIPTION
**Pull request by niwilso**
## Change Description

This PR enables the use of the allow list approach for image redaction (both standard images and DICOM images) and provides a demo notebook.

## Issue reference

This PR fixes issue #XX

## Checklist

- [X] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [ ] I have signed the CLA (if required)
- [X] My code includes unit tests
- [X] All unit tests and lint checks pass locally
- [X] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by niwilso** [comment]
Will merge from main after PR https://github.com/microsoft/presidio/pull/1113 is complete.

**Review by omri374 — APPROVED** [review]
Thanks! looks great. Do you want to keep it in draft until the other PRs are merged and the notebook is updated?

**Review by ayabel — APPROVED** [review]

**Review by SharonHart — APPROVED** [review]

**Comment by niwilso** [comment]
> Thanks! looks great. Do you want to keep it in draft until the other PRs are merged and the notebook is updated?

Thank you! Now that PR #1113 is merged in, I'll update the notebook in this PR and publish the PR. Likely sometime today.

**Review by SharonHart — APPROVED** [review]

**Review by omri374 — APPROVED** [review]

**Pull request merged**
